CONTRIBUTING.md

```diff
@@ -78,6 +79,66 @@ use only the standard libraries, or the third-party packages we have decided to
+A public function or method should only be used if using the Flux library would require the function or method to be public.
+
+### Comments and documentation
+Every new public function or method requires a valid GoDoc.
+Comments within the code should be written with proper capitalization and punctuation at the end of the sentence.
+
+### Updating markdown files for documentation
+The Flux team uses markdown for our documentation.
+When writing markdown, please use the following guidelines.
+
+- Each sentence in a paragraph is on its own line.
+- Use the `#` symbol for header sections.
+- Use the `-` symbol for bullet points.
+
+## Submitting a pull request
+To submit a pull request you should fork the Flux repository and make your change on a feature branch of your fork.
+Then generate a pull request from your branch against **master** of the Flux repository.
+Include in your pull request details of your change -- the **why** *and* the **how** -- as well as the testing you performed.
+Also, be sure to run the test suite with your change in place.
+Changes that cause tests to fail cannot be merged.
+The code you write must be unit tested within the same package as the code that you changed.
+
+There will usually be some back and forth as we finalize the change, but once that completes, it may be merged.
+When making changes based on the code review, please add new commits and please refrain from amending the original commit.
+This helps us keep track of the changes that have been made during the course of the review.
+
+When your change is ready to be merged, the Flux developer who is reviewing your code will approve the pull request and squash your commits into a single commit.
+The first commit in your changeset will be used as the final commit message.
+This first commit must follow the _conventional commit message_ format: https://www.conventionalcommits.org/en/v1.0.0-beta.3/.
+Please use this commit message format for commits that will be visible in influxdata/flux history.
+
+If your pull request only requires trivial changes before being approved and merged, the reviewer may make the change themselves and push to the branch in your fork.
+This makes it easier for us to fix a small nitpick such as formatting without requiring an additional back and forth.
+If the changes are non-trivial, we will always ask before changing any code.
+If you do not want us to do this at all under any circumstance, then you can disable this by unselecting **Allow edits from maintainers** in the pull request itself.
+
+To assist in review for the PR, please add the following to your pull request comment:
+
+```md
+- [ ] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)
+```
+
+### Signing the CLA
+In order to contribute back to Flux, you must sign the
+[InfluxData Contributor License Agreement](https://www.influxdata.com/legal/cla/) (CLA).
+
```

Comment:
This section should probably go before the PR section since they have to do this before opening the PR.